--- a/web/sites/all/themes/bootstrap/js/misc/autocomplete.js
+++ b/web/sites/all/themes/bootstrap/js/misc/autocomplete.js
@@ -30,10 +30,27 @@ Drupal.behaviors.autocomplete = {
  * Prevents the form from submitting if the suggestions popup is open
  * and closes the suggestions popup when doing so.
  */

 Drupal.autocompleteSubmit = function () {
-  return $('.form-autocomplete > .dropdown').each(function () {
+  var dropdownSelector = '.form-autocomplete > .dropdown';
+  var childrenSelector = dropdownSelector + ' li.active';
+  var $dropdown = $(dropdownSelector);
+  var $children = $(childrenSelector);
+  var noDropdown = $dropdown.length === 0;
+  var noChildren = $children.length === 0;
+
+  $dropdown.each(function () {
     this.owner.hidePopup();
-  }).length == 0;
+  });
+
+  return noChildren ? true : noDropdown ? true : false;
 };

 /**


