diff --git a/includes/views/handlers/taxonomy_entity_index_handler_argument_tid_depth.inc b/includes/views/handlers/taxonomy_entity_index_handler_argument_tid_depth.inc
index e66cb66..906c343 100644
--- a/includes/views/handlers/taxonomy_entity_index_handler_argument_tid_depth.inc
+++ b/includes/views/handlers/taxonomy_entity_index_handler_argument_tid_depth.inc
@@ -39,7 +39,7 @@ class taxonomy_entity_index_handler_argument_tid_depth extends views_handler_arg
     }
     // Now build the subqueries.
     $subquery = db_select('taxonomy_entity_index', 'tn');
-    $subquery->addField('tn', 'entity_id');
+    $subquery->addField('tn', 'revision_id');
     if (isset($this->base_table_info['table']['entity type'])) {
       $subquery->condition('entity_type', $this->base_table_info['table']['entity type']);
     }
diff --git a/includes/views/handlers/taxonomy_entity_index_handler_filter_tid_depth.inc b/includes/views/handlers/taxonomy_entity_index_handler_filter_tid_depth.inc
index b5de4b5..fa0a955 100644
--- a/includes/views/handlers/taxonomy_entity_index_handler_filter_tid_depth.inc
+++ b/includes/views/handlers/taxonomy_entity_index_handler_filter_tid_depth.inc
@@ -48,7 +48,7 @@ class taxonomy_entity_index_handler_filter_tid_depth extends views_handler_filte
 
     // Now build the subqueries.
     $subquery = db_select('taxonomy_entity_index', 'tei');
-    $subquery->addField('tei', 'entity_id');
+    $subquery->addField('tei', 'revision_id');
     if (isset($this->base_table_info['table']['entity type'])) {
       $subquery->condition('entity_type', $this->base_table_info['table']['entity type']);
     }
diff --git a/includes/views/taxonomy_entity_index.views.inc b/includes/views/taxonomy_entity_index.views.inc
index c62672f..74fd80c 100644
--- a/includes/views/taxonomy_entity_index.views.inc
+++ b/includes/views/taxonomy_entity_index.views.inc
@@ -12,14 +12,14 @@ function taxonomy_entity_index_views_data_alter(&$data) {
 
       // Join on all entity tables.
       $data['taxonomy_entity_index']['table']['join'][$base_table] = array(
-        'left_field' => $info['entity keys']['id'],
-        'field' => 'entity_id',
+        'left_field' => isset($info['entity keys']['revision']) ? $info['entity keys']['revision'] : $info['entity keys']['id'],
+        'field' => isset($info['entity keys']['revision']) ? 'revision_id' : 'entity_id',
       );
 
       $data[$base_table]['taxonomy_entity_index_tid_depth'] = array(
         'help' => t('Display content if it has the selected taxonomy terms, or children of the selected terms. Due to additional complexity, this has fewer options than the versions without depth.'),
         'group' => t('Taxonomy Entity Index'),
-        'real field' => $info['entity keys']['id'],
+        'real field' => isset($info['entity keys']['revision']) ? $info['entity keys']['revision'] : $info['entity keys']['id'],
         'argument' => array(
           'title' => t('Has taxonomy term ID on @entity_type (with depth and indexed in taxonomy_entity_index)', array('@entity_type' => $info['label'])),
           'handler' => 'taxonomy_entity_index_handler_argument_tid_depth',
@@ -36,10 +36,10 @@ function taxonomy_entity_index_views_data_alter(&$data) {
         'title' => t('Taxonomy terms on @entity_type', array('@entity_type' => $info['label'])),
         'help' => t('Relate @entity_type to taxonomy terms. This relationship will cause duplicated records if there are multiple terms.', array('@entity_type' => $info['label'])),
         'relationship' => array(
-          'real field' => $info['entity keys']['id'],
+          'real field' => isset($info['entity keys']['revision']) ? $info['entity keys']['revision'] : $info['entity keys']['id'],
           'label' => t('terms'),
           'base' => 'taxonomy_entity_index',
-          'base field' => 'entity_id',
+          'base field' => isset($info['entity keys']['revision']) ? 'revision_id' : 'entity_id',
           'extra' => array(
             array(
               'field' => 'entity_type',
